src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo;
+  loadingTodos: number[];
+  handleRemoveTodo: (id: number) => void;
+  handleToggleTodoStatus: (todo: Todo) => void;
+  handleUpdateTodoTitle: (todo: Todo, title: string) => Promise<void>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  loadingTodos,
+  handleRemoveTodo,
+  handleToggleTodoStatus,
+  handleUpdateTodoTitle,
+}) => {
+  const { id, title, completed } = todo;
+  const [newTitle, setNewTitle] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+  const todoInputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (isEditing && todoInputRef.current) {
+      todoInputRef.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleStartEditing = () => {
+    setIsEditing(true);
+  };
+
+  const handleCancelEditing = () => {
+    setNewTitle(title);
+    setIsEditing(false);
+  };
+
+  const handleSaveChanges = () => {
+    if (newTitle.trim() === '') {
+      handleRemoveTodo(id);
+    } else if (newTitle !== title) {
+      handleUpdateTodoTitle(todo, newTitle);
+    }
+
+    setIsEditing(false);
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Escape') {
+      handleCancelEditing();
+    } else if (event.key === 'Enter') {
+      handleSaveChanges();
+    }
```

Comment:
Before starting to work with functions and components, it would be very cool to define the variables that we use further into separate constants, so that we do not use magic values:
```
const ESCAPE= 'Escape';
const ENTER = 'Enter';
```
This will provide us with better readability of the code itself, conditions in the code and understanding that we will not accidentally make a mistake in a line.

Also, for better readability, it is not necessary to use the `if-else-if` construction:
```
if (event.key === ESCAPE) { ... }
if (event.key === ENTER ) { ... }
```